src/EnergyPlus/HVACManager.cc

```diff
@@ -2261,58 +2266,28 @@ void ReportAirHeatBalance(EnergyPlusData &state)
         state.afn->report();
     }
 
     // Reports zone exhaust loss by exhaust fans
-    for (int ZoneLoop = 1; ZoneLoop <= state.dataGlobal->NumOfZones; ++ZoneLoop) { // Start of zone loads report variable update loop ...
-        auto &zone = state.dataHeatBal->Zone(ZoneLoop);
-        auto &znAirRpt = state.dataHeatBal->ZnAirRpt(ZoneLoop);
-        auto &zoneEquipConfig = state.dataZoneEquip->ZoneEquipConfig(ZoneLoop);
-        CpAir = Psychrometrics::PsyCpAirFnW(state.dataEnvrn->OutHumRat);
-        H2OHtOfVap = Psychrometrics::PsyHgAirFnWTdb(state.dataEnvrn->OutHumRat, zone.OutDryBulbTemp);
-        ADSCorrectionFactor = 1.0;
-        if (state.afn->simulation_control.type == AirflowNetwork::ControlType::MultizoneWithDistributionOnlyDuringFanOperation) {
-            if ((state.dataZoneEquip->ZoneEquipAvail(ZoneLoop) == Avail::Status::CycleOn ||
-                 state.dataZoneEquip->ZoneEquipAvail(ZoneLoop) == Avail::Status::CycleOnZoneFansOnly) &&
-                state.afn->AirflowNetworkZoneFlag(ZoneLoop)) {
-                ADSCorrectionFactor = 0.0;
-            }
-        }
-
-        znAirRpt.ExhTotalLoss = 0;
-        znAirRpt.ExhSensiLoss = 0;
-
-        for (int FanNum = 1; FanNum <= (int)state.dataFans->fans.size(); ++FanNum) {
-            auto const *thisFan = state.dataFans->fans(FanNum);
-            //  Add reportable vars
-            if (thisFan->type == HVAC::FanType::Exhaust) {
-                for (int ExhNum = 1; ExhNum <= zoneEquipConfig.NumExhaustNodes; ExhNum++) {
-                    if (thisFan->inletNodeNum == zoneEquipConfig.ExhaustNode(ExhNum)) {
-                        znAirRpt.ExhTotalLoss +=
-                            thisFan->outletAirMassFlowRate * (thisFan->outletAirEnthalpy - state.dataEnvrn->OutEnthalpy) * ADSCorrectionFactor;
-                        znAirRpt.ExhSensiLoss +=
-                            thisFan->outletAirMassFlowRate * CpAir * (thisFan->outletAirTemp - zone.OutDryBulbTemp) * ADSCorrectionFactor;
-                        break;
-                    }
-                }
-            }
+    for (int zoneNum = 1; zoneNum <= state.dataGlobal->NumOfZones; ++zoneNum) { // Start of zone loads report variable update loop ...
+        auto &zone = state.dataHeatBal->Zone(zoneNum);
+        auto &znAirRpt = state.dataHeatBal->ZnAirRpt(zoneNum);
+        auto &znEquipConfig = state.dataZoneEquip->ZoneEquipConfig(zoneNum);
+        reportAirHeatBal1(state, znAirRpt, znEquipConfig, zoneNum);
+    }
+    if (state.dataHeatBal->doSpaceHeatBalance) {
+        for (int spaceNum = 1; spaceNum <= state.dataGlobal->numSpaces; ++spaceNum) {
+            auto &spAirRpt = state.dataHeatBal->spaceAirRpt(spaceNum);
+            auto &spEquipConfig = state.dataZoneEquip->spaceEquipConfig(spaceNum);
+            int zoneNum = state.dataHeatBal->space(spaceNum).zoneNum;
+            reportAirHeatBal1(state, spAirRpt, spEquipConfig, zoneNum, spaceNum);
```

Comment:
Break up `ReportAirHeatBalance` into smaller functions that can be re-used for zones and spaces.